packages/openactive-broker-microservice/src/core.js

```diff
@@ -904,7 +834,7 @@ async function storeChildOpportunityItem(item) {
  * Merge a child- and parent- opportunity and then save them to the
  * criteria-oriented cache and notify any listeners.
  *
- * @param {import('./models/core').OpportunityItemRow} row
+ * @param {Readonly<import('./models/core').OpportunityItemRow>} row
```

Comment:
Make input data readonly to be sure that they aren't secretly doing mutations (which would not work as expected if data is being stored in something outside of JS memory)